### PR TITLE
config: Add --whitelist option

### LINKED ISSUE #351: Add whitelisting of networks.

### PR DESCRIPTION
**Pull request by dajohi**
The whitelist option is used to specify networks that will not be
banned.

Fixes #351

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -101,6 +101,7 @@ type config struct {
 	DisableBanning      bool          `long:"nobanning" description:"Disable banning of misbehaving peers"`
 	BanDuration         time.Duration `long:"banduration" description:"How long to ban misbehaving peers.  Valid time units are {s, m, h}.  Minimum 1 second"`
 	BanThreshold        uint32        `long:"banthreshold" description:"Maximum allowed ban score before disconnecting and banning misbehaving peers."`
+	Whitelists          []string      `long:"whitelist" description:"Add a network that will not be banned. (eg. 192.168.1.1/32)"`
```
Comment (jrick) [inline config.go]: only works with CIDR or can just an IP address also be used?

**Comment by cjepson** [comment]
Does it work with IPv6 too?

**Comment by dajohi** [comment]
@jrick CIDR only the way it is now.  I could add single IP support if desired.

**Comment by dajohi** [comment]
@cjepson yes

**Comment by cjepson** [comment]
@dajohi Would prefer single IP support for ease of use.

**Comment by dajohi** [comment]
This will be synced with btcd - https://github.com/btcsuite/btcd/pull/769

**Comment by jcvernaleo** [comment]
@dajohi so if this is planned on going in as part of a btcd sync after the similar commit goes into btcd maybe this PR should be closed (since it should not be accepted directly), right?

**Comment by dajohi** [comment]
leaving open til the code is in dcrd

**Comment by alexlyp** [comment]
Still leaving open for upcoming btcd sync that should include this.

File: config.go (hunk 1 above)
Comment (JasonSome) [inline config.go]: @dajohi

**Comment by jrick** [comment]
Since we are backporting features we want from btcd instead of waiting for the sync, i'm ok with this going in.

It has merge conflicts however and needs a rebase.

**Comment by davecgh** [comment]
The upstream version is waiting for updates as well.  This is not ready to go in as is.

**Comment by dajohi** [comment]
@jrick single ip support is in https://github.com/btcsuite/btcd/pull/769

**Comment by davecgh** [comment]
All concerns have been addressed.

**Pull request merged**